### PR TITLE
Velocity Estimation

### PR DESCRIPTION
**Pull request by d-lowl**
* Incremental PCA
* Velocity calculation in a notebook
* Velocity and distance estimate in the Android app
* Unit test for metric calculations in the Android app

### CODE REVIEW

File: android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/MainActivity.kt [hunk 1]
```diff
@@ -149,13 +162,17 @@ class MainActivity : AppCompatActivity() {
                     sessionService?.reset()
                 }
                 BluetoothLeService.ACTION_GATT_DISCONNECTED -> {
-                    finish()
+//                    finish()
```
Comment (archena) [inline android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/MainActivity.kt:165]: Why commented out?

File: android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/math/math.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package ai.fuzzylabs.wearablemyfoot.math
+
+fun cumtrapz(y: Iterable<Double>, x: Iterable<Double>? = null): Sequence<Double> {
```
Comment (archena) [inline android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/math/math.kt:3]: Could you add some tests for this one?

File: android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/imu/IMUSessionService.kt [hunk 3]
```diff
@@ -116,18 +129,25 @@ class IMUSessionService : Service() {
...
-        } else {
-            0.0
-        }
-    }
+    fun getCurrentCadence(): Double = session.currentElement.cadence
+
+    /**
+     * Get current speed in kmh
+     */
+    fun getCurrentSpeed(): Double = session.currentElement.speed * 3.6
```
Comment (archena) [inline android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/imu/IMUSessionService.kt:137]: What is `3.6` for here? may be better as a named constant

File: android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/MainActivity.kt (hunk 1 above)
Comment (d-lowl) [reply android/app/src/main/java/ai/fuzzylabs/wearablemyfoot/MainActivity.kt:165]: It shouldn't be. Was trying to find a bug (dealing with concurrency). Uncommented now.

**Review by archena — APPROVED** [review]
Looks good!

**Pull request merged**